### PR TITLE
[Dance] Fix visualization overflow on initial page load for small screens

### PR DESCRIPTION
**Pull request by maddiedierker**
Fixes a bug where the song selector and game buttons horizontally-overflowed in Dance Party on certain screen sizes (browser width > 1150 && height > 900).

There is also some clean-up in this PR. The actual (short-term) fix for this bug is in commit [531826d](https://github.com/code-dot-org/code-dot-org/pull/41454/commits/531826dc2390478d2819b6cb2b6a25b10a40b26c). The long-term fix is logged as follow-up work.

**Before:**
Screen Shot 2021-06-15 at 1 46 43 PM

**After:**

https://user-images.githubusercontent.com/9812299/124681454-6d3e2680-de7d-11eb-9f5c-d77d4600c55c.mov

## Links

- [STAR-1606](https://codedotorg.atlassian.net/browse/STAR-1606)

## Testing story

Relies on existing UI/eyes tests.

## Follow-up work

Right now, StudioApp controls resizing logic imperatively via [StudioApp.resizeVisualization](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/StudioApp.js#L1524-L1596). I've logged a follow-up task to refactor our approach to resizing as [STAR-1621](https://codedotorg.atlassian.net/browse/STAR-1621).

### CODE REVIEW

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 1]
```diff
@@ -162,6 +143,26 @@ class DanceVisualizationColumn extends React.Component {
...
+    height: GAME_HEIGHT
+  },
+  container: {
+    touchAction: 'none',
+    background: '#fff',
+    position: 'relative',
+    overflow: 'hidden'
+  },
+  loadingContainer: {
+    // The value of display will be controlled externally by StudioApp.
```
Comment (ajpal) [inline apps/src/dance/DanceVisualizationColumn.jsx]: nit: "is controlled" instead of "will be controlled" feels slightly more clear (to me at least)

**Review by ajpal — COMMENTED** [review]
thanks for doing the cleanup as well!

File: apps/src/dance/Dance.js [hunk 2]
```diff
@@ -131,6 +131,11 @@ Dance.prototype.init = function(config) {
     this.studioApp_.init(config);
     this.studioAppInitPromiseResolve();
 
+    // Manually trigger a 'resize' after the relevant React components have mounted
+    // to ensure consistent sizing/scaling and prevent overflow after instructions
+    // have rendered.
+    this.studioApp_.resizeVisualization();
```
Comment (breville) [inline apps/src/dance/Dance.js]: aside: hopefully one day we can simplify this behaviour and use React top to bottom!

Comment (maddiedierker) [reply apps/src/dance/Dance.js]: yes, hopefully soon! i have to flesh out the details, but that work is tracked as [STAR-1621](https://codedotorg.atlassian.net/browse/STAR-1621)

**Comment by maddiedierker** [comment]
@ajpal i made a couple of changes to my fix here -- could you take another look?

**Review by ajpal — APPROVED** [review]

**Pull request merged**